blivet/fstab.py

```diff
@@ -0,0 +1,279 @@
+        spec = getattr(device, 'fstab_spec', None)
+
+        file = None
+        if device.format.mountable:
+            file = device.format.mountpoint
+        if device.format.type == 'swap':
+            file = 'swap'
+
+        vfstype = device.format.type
+        mntops = None
+
+        return spec, file, vfstype, mntops
+
+    def add_entry(self, spec, file, vfstype, mntops, freq=None, passno=None, comment=None):
+        # Default mount options
+        if mntops is None:
+            mntops = 'defaults'
+
+        # Whether the fs should be dumped by dump(8), defaults to 0 
+        if freq is None:
+            freq = 0
+
+        # Order of fsck run at boot time. '/' should have 1, other 2, defaults to 0
+        if passno is None:
+            if file is None:
+                file = ''
+            if file == '/':
+                passno = 1
+            elif file.startswith('/boot'):
+                passno = 2
+            else:
+                passno = 0
+
+        entry = Fs()
+
+        entry.source = spec
+        entry.target = file
+        entry.fstype = vfstype
+        entry.append_options(mntops)
+        entry.freq = freq
+        entry.passno = passno
+
+        if comment is not None:
+            # add '# ' at the start of any comment line and newline to the end of comment
+            modif_comment = '# ' + comment.replace('\n', '\n# ') + '\n'
+            entry.comment = modif_comment
+
+        self._table.add_fs(entry)
+
+    def remove_entry(self, spec, file, vfstype, mntops):
+        fs = self.find(spec, file, vfstype, mntops)
+        if fs:
+            self._table.remove_fs(fs)
+
+    def write(self, dest_file = None):
+        if dest_file is None:
+            dest_file = self.dest_file
+
+        if os.path.exists(dest_file):
+            self._table.replace_file(dest_file)
+        else:
+            # write will fail if underlying directories do not exist
+            self._table.write_file(dest_file)
+
+    def find(self, spec, file, vfstype, mntops):
+        # TODO add find_pair if both spec and file are given
+
+        entry = None
+        if spec is not None:
+            try:
+                entry = self._table.find_source(spec, MNT_ITER_FORWARD)
+            except LibmountException:
+                entry = None
+
+        if file is not None:
+            try:
+                entry = self._table.find_target(file)
+            except LibmountException:
+                entry = None
+        return entry
+
+    def update(self, devices):
+        # remove entries not tied to any device
+
+        fstab_entries = []
+        entry = self._table.next_fs()
+        while entry:
+            fstab_entries.append(entry)
+            entry = self._table.next_fs()
+
+        for device in devices:
+            args = self.from_device(device)
+            entry = self.find(*args)
+
+            # remove from the list if found
+            try:
+                fstab_entries.remove(entry)
```

Comment:
I might be missing something here, but `fstab_entries` isn't used after this so this part actually doesn't remove the devices from the table?

Reply:
Fixed.